From ca973c36a7579bce356f057c2d0302e10d43666e Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 17 Feb 2022 15:21:15 +0100
Subject: [PATCH 216/529] dt-bindings: rtc: stm32: add alarm A out property to
 select output

STM32 RTC can pulse some SOC pins when an alarm of RTC expires.

This patch adds property to activate alarm A. The pulse can output on
three pins RTC_OUT1, RTC_OUT2, RTC_OUT2_RMP (PC13, PB2, PI8 on stm32mp15)
(PC13, PB2, PI1 on stm32mp13).

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I77f7649841f7d73883bce8915e397d0592ba19b3
---
 .../devicetree/bindings/rtc/st,stm32-rtc.yaml | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
+++ b/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
@@ -59,6 +59,13 @@ properties:
       Refer to <include/dt-bindings/rtc/rtc-stm32.h> for the supported values.
       Pinctrl state named "default" may be defined to reserve pin for RTC output.
 
+  st,alarm:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: |
+      To select and enable RTC Alarm A output.
+      Refer to <include/dt-bindings/rtc/rtc-stm32.h> for the supported values.
+      Pinctrl state named "default" may be defined to reserve pin for RTC output.
+
 allOf:
   - if:
       properties:
@@ -75,6 +82,9 @@ allOf:
         st,lsco:
           maxItems: 0
 
+        st,alarm:
+          maxItems: 0
+
         clock-names: false
 
       required:
@@ -95,6 +105,9 @@ allOf:
         st,lsco:
           maxItems: 0
 
+        st,alarm:
+          maxItems: 0
+
       required:
         - clock-names
         - st,syscfg
@@ -117,6 +130,9 @@ allOf:
         st,lsco:
           maxItems: 1
 
+        st,alarm:
+          maxItems: 1
+
       required:
         - clock-names
 
@@ -153,8 +169,9 @@ examples:
       clocks = <&rcc RTCAPB>, <&rcc RTC>;
       clock-names = "pclk", "rtc_ck";
       interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
+      st,alarm = <RTC_OUT1>;
       st,lsco = <RTC_OUT2_RMP>;
-      pinctrl-0 = <&rtc_out2_rmp_pins_a>;
+      pinctrl-0 = <&rtc_out1_pins_a &rtc_out2_rmp_pins_a>;
       pinctrl-names = "default";
     };
 
